packages/require/lib/types.ts

```diff
@@ -0,0 +1,112 @@
+import TruffleConfig from "@truffle/config";
+import Resolver from "@truffle/resolver";
+import TruffleError from "@truffle/error";
+
+export interface RequireOptions {
+  /**
+   * The path to the source file to be required/executed
+   */
+  file: string;
+
+  /**
+   * The `@truffle/config` for the user's project
+   */
+  config: TruffleConfig;
+}
+
+export interface ExecOptions {
+  contracts_build_directory: string;
+  /**
+   * Path to file to execute. Must be a module that exports a function.
+   */
+  file: string;
+
+  /**
+   * Object containing any global variables you'd like set when this function is
+   * run.
+   */
+  context: {
+    [index: string]: any;
+  };
+
+  /**
+   * The instance of `@truffle/resolver` to be passed into the global context
+   * when executing the script
+   */
+  resolver: Resolver;
+
+  /**
+   * The JSON RPC provider that will be used by the interface-adpater and web3
+   * instance injected into the script
+   */
+  provider: Provider;
+
+  /**
+   * A reference to a network defined in the user's `@truffle/config`.
+   */
+  network: string | number;
+
+  /**
+   * The id of the network as reported by the `net_version` JSON RPC method.
+   */
+  network_id: string | number;
```

Comment:
The JSON-RPC type is an integer returned as a string; I don't think `number` belongs here.

Maybe something in truffle converts it to a number? I don't believe it should be normalized to `number`, as the value may be greater than `MAX_SAFE_INTEGER`.

Reply:
The source of this is usually the user's `truffle-config.js` file. [Examples in our documentation](https://trufflesuite.com/docs/truffle/reference/configuration/#networks) specify `network_id` as a number in `truffle-config.js`, and I suspect there is quite a lot of code in the wild that does this.